### PR TITLE
Implemented Locator

### PR DESCRIPTION
**Pull request by XedinUnknown**
Abstract and File-based.

### CODE REVIEW

File: src/ConfigSourceAwareTrait.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+namespace Dhii\Modular\Locator;
+
+/**
+ * Functionality that allows setting and retrieval of a module configuration source (protected).
+ *
+ * @since [*next-version*]
+ */
+trait ConfigSourceAwareTrait
+{
+    /**
```
Comment (mecha) [inline src/ConfigSourceAwareTrait.php:12]: Missing `configSource` property.

File: src/AbstractModuleLocator.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     *
+     * @since [*next-version*]
+     *
+     * @return mixed[] A list of module configurations.
+     */
+    protected function _locate()
+    {
+        $configs = [];
+        foreach ($this->_getSources() as $_source) {
+            $configs[md5($_source)] = $this->_read($_source);
```
Comment (mecha) [inline src/AbstractModuleLocator.php]: The hashing done here should be in a separate method. I assume that this is what `_generateKeyFromSource()` was created for. Perhaps you forgot to change this line?

**Review by mecha — CHANGES_REQUESTED** [review]

File: src/ConfigSourceAwareTrait.php (hunk 1 above)
Comment (XedinUnknown) [reply src/ConfigSourceAwareTrait.php:12]: Was trying to avoid constraints to the property signature. But you are right, the stricter - the better.

**Review by mecha — APPROVED** [review]
LGTM.

**Pull request merged**